libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/PrettyCore.hs

```diff
@@ -1338,19 +1338,28 @@ ppConwayUtxoPredFailure = \case
       [ ("provided interval", ppValidityInterval vi)
       , ("current slot", pcSlotNo slot)
       ]
-  ConwayRules.MaxTxSizeUTxO actual maxs ->
-    ppRecord "MaxTxSizeUTxO" [("Actual", ppInteger actual), ("max transaction size", ppInteger maxs)]
+  ConwayRules.MaxTxSizeUTxO mm@(Mismatch _ _) ->
+    let Mismatch {..} = mm
```

Comment:
```suggestion
  ConwayRules.MaxTxSizeUTxO Mismatch {..} ->
```

I noticed this in quite a few other places, but haven't commented on them in case there's a reason.

I'm also wondering whether some of the code like this

```
         PPViewHashesDontMatch mm@(Mismatch _ _) ->
            Sum PPViewHashesDontMatch 13
              !> E (\(Mismatch {..}) -> encCBOR mismatchSupplied <> encCBOR mismatchExpected) mm
 ```

could be simplified in the same way, but I don't know enough about the CBOR library to be sure. For example,

```
         PPViewHashesDontMatch Mismatch {..} ->
            Sum PPViewHashesDontMatch 13 !> To mismatchSupplied !> To mismatchExpected
```

Reply:
The reason this will not work is because of the types. The `Coders` module is a good place to look into how the `!>/<!` `Sum/SumD` duals work to make sure that the type we are serde-ing gets done correctly without mistakes. Using just one `!>` and not two, like in your example, we make the type inference happy that `PPViewHashesDontMatch` satisfies the kind `* -> *`, as opposed to the kind `* -> * -> *` that the double `!>` in your example would imply.

Reply:
Suggestion from @neilmayhew in this particular case is a legitimate one. However, the only reason why it is safe to rely on `RecordWildCards` is because we know that `Mismatch` can ever only have two fields, otherwise we would not be able to use it any more in any of the predicate failures.
```suggestion
  ConwayRules.MaxTxSizeUTxO Mismatch {..} ->
```

@neilmayhew for all other types that use `RecordWildCards` this trick:

```haskell
fun foo@(Foo _ _ _) = 
  let Foo {..} = foo
   in
```
Ensures that the compiler will notify us about all the places that needed to be adjusted whenever we add new fields to the `Foo` type. Without this trick `RecordWildCards` is dangerous to use.

With respect to "coders", it has other restrictions that mess with type inference, as @aniketd pointed out.

Reply:
That being said, I really don't care about the Pretty instances, because my hope is that we can completely get rid of them in favor of `ToExpr` a not so distant future. Therefore, @aniketd, feel free to leave your changes as is in this module.